packages/platform-sdk/src/signatories/signatory.ts

```diff
@@ -0,0 +1,124 @@
+import { ForbiddenMethodCallException } from "../exceptions";
+import { MnemonicSignatory } from "./mnemonic";
+import { MultiMnemonicSignatory } from "./multi-mnemonic";
+import { MultiSignature, MultiSignatureSignatory } from "./multi-signature";
+import { PrivateKeySignatory } from "./private-key";
+import { SecondaryMnemonicSignatory } from "./secondary-mnemonic";
+import { SecondaryWIFSignatory } from "./secondary-wif";
+import { SenderPublicKeySignatory } from "./sender-public-key";
+import { SignatureSignatory } from "./signature";
+import { WIFSignatory } from "./wif";
+
+type SignatoryType = MnemonicSignatory
+	| MultiMnemonicSignatory
+	| SecondaryMnemonicSignatory
+	| WIFSignatory
+	| SecondaryWIFSignatory
+	| PrivateKeySignatory
+	| SignatureSignatory
+	| SenderPublicKeySignatory
+	| MultiSignatureSignatory;
+
+export class Signatory {
+	readonly #signatory: SignatoryType;
+
+	public constructor(signatory) {
+		this.#signatory = signatory;
+	}
```

Comment:
There used to be a TS syntax for simplifying this pattern
```typescript
class X {
  private a: A;
  private b: B;
  
  public constructor(a: A, b: B) {
    this.a = a;
    this.b = b;
  }
}
```
to just this:
```typescript
class X {
  public constructor(private a: A, private b: B) {}
}
```

They haven't made an equivalent for the new #private variables?

Reply:
Not that I'm aware.